### ISSUE TITLE
Using logcumsumexp in loss calculation

### ISSUE THREAD

**Issue by avihu111** [havakv/pycox#144]
Hi!
Came across this wonderful package and I have a small idea for improvement:
In here:
https://github.com/havakv/pycox/blob/0e9d6f9a1eff88a355ead11f0aa68bfb94647bf8/pycox/models/loss.py#L422
you implement the logcumsumexp function manually, while pytorch has a built-in implementation:
https://pytorch.org/docs/stable/generated/torch.logcumsumexp.html

I thought it might be worth replacing to their implementation.
Cheers!

**Comment by havakv** [comment]
Thanks! No reason to implement this then. Do you want to make a PR?
